aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -29,12 +29,22 @@ void sub_kernel(TensorIterator& iter, Scalar alpha_scalar) {
   add_kernel(iter, -alpha_scalar);
 }
 
+template <typename T>
+T multiply(T a, T b) {
```

Comment:
Try to avoid this. `*` should work on `bool`. You might need to add an overload to Vec256.

Reply:
It would work without this but the build on CircleCi is failing due to an error about bad usage of "*" with two bools.

Reply:
Then suppress the warning. I'm not a fan of unnecessarily verbose code driven by the compiler's preferences.

Reply:
@cpuhrsch, made a good argument against adding an overload to Vec256.
Also, i dont feel comfortable with fully blocking warning for cases like this. If there was a way to block it for this method, i would be fine but as far as i can tell - we block gcc warnings for the whole project and i dont think its a good way to go. 
Please, let me know if you have strong preferences here

Reply:
```
#pragma GCC diagnostic ignored "-Wint-in-bool-context"
```

We should stick to the same operators because it keeps the code simple and **reduces** the chances of incorrect behavior. For example, this PR previously had an incorrect definition of `-` because it didn't use `operator-` -- it was the opposite of how `-` operates on a bool in both C/C++ and Python.